opm/simulators/timestepping/TimeStepControl.cpp

```diff
@@ -152,6 +153,12 @@ namespace Opm
                 std::cout << "Computed step size (tol): " << unit::convert::to( newDt, unit::day ) << " (days)" << std::endl;
             return newDt;
         }
+        else if ( error == 0.)
+        {
+            if ( verbose_ )
+                std::cout << "The solution between time step does not change, there is no time step constraint from the PID time step control " << std::endl;
+            return std::numeric_limits<double>::max();
+        }
         else
         {
             // values taking from turek time stepping paper
```

Comment:
For here, if `error==0`, `error_[2]` will be zero here. For more complicated scenarios, `errors_[1]` can be zero, the `else` branch will also fail.